pkg/kubelet/pod/pod_manager.go

```diff
@@ -153,6 +158,9 @@ func (pm *basicManager) UpdatePod(pod *v1.Pod) {
 // lock.
 func (pm *basicManager) updatePodsInternal(pods ...*v1.Pod) {
 	for _, pod := range pods {
+		if pm.secretManager != nil {
```

Comment:
Where does the unregistering of the old version happen? To me it looks like this would keep incrementing each secret on an update?